### PR TITLE
ignore menu events sent when a modal is shown

### LINKED ISSUE #3308: keyboard events from app menu still triggered when displaying modal
### Description

Keyboard events still triggering in modal

### Version

**GitHub Desktop version:** 1.0.9-beta0

**OS version:** Windows

### Steps to Reproduce

1. Open repository settings dialog
2. Press CTRL-B

**Expected behavior:** Keyboard event should be ignored

**Actual behavior:** Branch switcher opens in background

image

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #3308

A simple-enough fix without going down the rabbit hole of disabling the menu completely when a modal is displayed. I'm not aware of cases where we might need to handle these events, so shout if this makes you feel uncomfortable.

### CODE REVIEW

**Comment by joshaber** [comment]
On macOS there's visual feedback in the menu bar when you perform a shortcut. I worry it'd be confusing if I got that feedback but the shortcut didn't actually work 🤔

**Comment by shiftkey** [comment]
> On macOS there's visual feedback in the menu bar when you perform a shortcut.

Yeah, that's exactly what I was afraid of. There doesn't seem to be an Electron API for disabling the menu globally, so I guess if we want to do this Properly™ we need to do more IPC work based on when a modal is shown or dismissed...

**Comment by shiftkey** [comment]
Dropping the review label to size up how complicated a proper solution will look.

**Comment by j-f1** [comment]
How about toggling the `disabled` key on all of the menu items? I’m pretty sure that’s what native dialogs do.

**Comment by shiftkey** [comment]
I've got mixed feelings about this approach, but I'll sleep on it and see if anyone else has feels about it...

File: app/src/main-process/main.ts [hunk 1]
```diff
@@ -188,6 +188,18 @@ app.on('ready', () => {
     }
   )
 
+  function setAllMenuItems(enabled: boolean) {
```
Comment (shiftkey) [inline app/src/main-process/main.ts]: Probably the part that makes me most uncomfortable about this change is that we might re-enable the whole menu when the modal is dismissed. After more testing on macOS I see this re-enables all menu items when you have no repositories selected (including features you shouldn't have access to).

I'll see if this is easy to trigger or if it requires tracking the current menu state _somehow_...

File: app/src/lib/menu-update.ts [hunk 2]
```diff
@@ -80,7 +80,50 @@ function menuItemStateEqual(state: IMenuItemState, menuItem: MenuItem) {
   return true
 }
 
+const allMenuIds: ReadonlyArray<MenuIDs> = [
```
Comment (shiftkey) [inline app/src/lib/menu-update.ts]: This is probably the only janky part of the change - I'd love to keep this in sync with the `MenuIDs` source but I won't hold up this PR unless someone has some bright ideas to solve this with TYPES!

Comment (joshaber) [reply app/src/lib/menu-update.ts]: Yeah, less than ideal but I don't know of a better way 🤔

**Comment by tierninho** [comment]
So we are disabling _all_ menu interaction if _any_ modal is open?

**Comment by shiftkey** [comment]
> So we are disabling _all_ menu interaction if any modal is open?

To clarify, we disable all the Desktop-specific menu items when a modal is open. There's some macOS default menus which remain unaffected by this:











I didn't really look into the Windows side, as that menu isn't interactive when you have a modal open...

**Pull request merged**